gpii/node_modules/flowManager/src/UserLogonStateChange.js

```diff
@@ -115,49 +142,81 @@ fluid.defaults("gpii.flowManager.userLogonStateChange.matchMakingStateChangeHand
+ * @return {Boolean} Returns true if the proximityTrigger should be ignored (i.e. is considered a bounce). This is done according
+ *              to the following rule: any RFID actions is ignored for <mytoken> if a login/logout for <mytoken> is in progress
+ *              OR if the last login/logout process for <mytoken> finished less than {debounceTimeMs} milliseconds ago
+ */
+gpii.flowManager.userLogonHandling.isBounce = function (lifecycleManager, debounceTimeMs, userToken) {
+    // if login/logout is in progress for <userToken> we consider it a bounce
+    var now = Date.now();
+    var model = lifecycleManager.model;
+    if (userToken === model.logonChange.userToken) {
+        return model.logonChange.inProgress || now - lifecycleManager.model.logonChange.timeStamp < debounceTimeMs;
+    } else {
+        return false;
+    }
+};
+
 /**
- * /user/:userToken/logonChange handler
- * * If no user is logged in, user <userToken> will be logged in
- * * If <userToken> is logged in, <userToken> will be logged out
- * * If another user is logged in, nothing happens
- * If "reset" is used as the <mytoken> for any of the above URLs, any currently logged in user will be logged out
+ * /user/:userToken/proximityTriggered handler
+ * RFID with <mytoken>:
+ * * Debounce rule: any RFID actions is ignored for <mytoken> if a login/logout for <mytoken> is in progress
+ * *                OR if the last login/logout process for <mytoken> finished less than 5 seconds ago
+ * * If no user is logged in and debounce doesn't apply, log in <mytoken>
+ * * If <mytoken> is logged in and debounce doesn't apply, log out <mytoken>
+ * * If another user is already logged in or in the process of logging in or out, log that user out and log in <mytoken>
  */
-gpii.flowManager.userLogonStateChange.handleRequest = function (userToken, lifecycleManager, that) {
+gpii.flowManager.userLogonHandling.handleProximityTriggerRequest = function (that, lifecycleManager, userToken) {
+    fluid.log("proximity triggered by token: ", userToken);
     // find currently logged in user:
     var activeTokens = lifecycleManager.getActiveSessionTokens();
+
     if (userToken === "reset") {
-        gpii.flowManager.userLogonStateChange.handleResetToken(activeTokens[0], that);
+        gpii.flowManager.userLogonHandling.handleResetToken(activeTokens[0], that);
         return;
     }
+
+    if (that.isBounce(userToken, that)) {
+        fluid.log("Proximity trigger ignored due to bounce rules");
+        that.errorResponse("Proximity trigger ignored due to bounce rules. Please wait current logon change is complete", 429);
+        return;
+    }
+
     if (activeTokens.length === 0) { // if no user is logged in
         that.loginUser(userToken);
         return;
     }
 
+    // if a different user is logged in, log that user out and log the new user in
     if (activeTokens.length === 1) {
         if (activeTokens[0] === userToken) {
             that.logoutUser(activeTokens[0]);
```

Comment:
I worry more about race conditions with this pull request - not that we didn't have them before, but we had simply decided to ignore all issues relating to closely spaced logins/logouts. Now we are considering them, we have to make sure that we at least aren't making the existing situation worse.
I see (at least) two problems here - firstly, that this implementation doesn't agree with the comment - we don't actually make any attempts to log the new user in after we log the existing one out. This means that the test named "Login with a different user with proximity trigger should log previous user out" must somehow be faulty, but I can't immediately see how.
Secondly, the logoutUser operation is actually asynchronous, so even if you had logged out successfully, there's no clarity that you wouldn't corrupt the system state by overlapping the logout process. 
There are other possible race conditions - I think we are saved from some of them by virtue of Infusions' "event nulling" that will prevent the "onMatchDone" event of a completed request from firing (e.g. if the user logs out before they complete logging on). But, for example, there is also the following branch in the LifecycleManager where we may interfere with ourselves: https://github.com/kaspermarkus/universal/blob/1e4771f522dc544b9f6b47c64698793b99478d5b/gpii/node_modules/lifecycleManager/src/LifecycleManager.js#L557